postage_stamps.py

```diff
@@ -0,0 +1,272 @@
+
+import numpy as np
+from astropy.table import Table, vstack
+from tqdm import tqdm
+from astropy.coordinates import SkyCoord
+from astropy import units as u
+
+import GCRCatalogs
+import lsst.afw.geom as afw_geom
+import lsst.daf.persistence as dafPersist
+from lsst.pex.exceptions import LengthError
+
+
+def get_valid_dataids(butler, **kwargs):
+    """Return all dataId values that exist
+
+    Args:
+        butler (Butler): A data access butler
+
+    Returns:
+        A list of dataID values
+    """
+
+    subset = butler.subset('deepDiff_diaSrc')
+    id_list = [dr.dataId for dr in tqdm(subset) if dr.datasetExists()]
+    if not id_list:
+        raise RuntimeError('No valid dataId found.')
+    return id_list
+
+
+def get_truth_catalog():
+    """Load catalog data from "dc2_truth_run1.2_variable_summary"
+
+    Returns:
+       A SourceCatalog object with id, coord_ra, and coord_dec fields
+    """
+
+    # Load truth data
+    truth_gcr = GCRCatalogs.load_catalog('dc2_truth_run1.2_variable_summary')
+    truth_data = truth_gcr.get_quantities(['uniqueId', 'ra', 'dec'])
+
+    truth_table = Table(
+        data=[truth_data['uniqueId'], truth_data['ra'], truth_data['dec']],
+        names=['truth_id', 'truth_ra', 'truth_dec'],
+        dtype=[np.int64, float, float]
+    )
+
+    truth_table['truth_ra'].unit = u.degree
+    truth_table['truth_dec'].unit = u.degree
+    return truth_table
+
+
+def get_diasrc_catalog(butler, data_id):
+    """For a collection of data Ids, return a table of all DIA sources
+
+    Args:
+        butler (Butler): A data access butler
+        data_id  (dict): A dictionary of values uniquely identifying an image
+
+    Returns:
+       An astropy table
+    """
+
+    diasrc_table = Table(
+        names=['src_id', 'src_ra', 'src_dec', 'visit', 'filter',
+               'detector'],
+        dtype=[np.int64, float, float, np.int64, str, 'U10']
+    )
+
+    visit = data_id['visit']
+    filter_ = data_id['filter']
+    detector = data_id['detector']
+    for source in butler.get('deepDiff_diaSrc', dataId=data_id):
+        id_ = source['id']
+        ra = source['coord_ra']
+        dec = source['coord_dec']
+        diasrc_table.add_row([id_, ra, dec, visit, filter_, detector])
+
+    diasrc_table['src_ra'].unit = u.rad
+    diasrc_table['src_dec'].unit = u.rad
+    return diasrc_table
+
+
+def match_dataid(butler, data_id, truth_ctlg, radius):
+    """Cross match DIA sources from a given data Id with the truth catalog
+
+    Args:
+        butler     (Butler): A data access butler
+        data_id      (dict): A dictionary of values uniquely identifying an image
+        truth_ctlg  (Table): GCR truth catalog
+        radius      (float): Match radius in arc-seconds (Default: 1)
+
+    Returns:
+        Cross match results as an astropy table
+    """
+
+    source_ctlg = get_diasrc_catalog(butler, data_id)
+    source_skyc = SkyCoord(
+        ra=source_ctlg['src_ra'], dec=source_ctlg['src_dec'])
```

Comment:
I would have just called these columns `ra`, `dec` above.
Or `coord_ra`, `coord_ded`, and specify `units=u.rad`  (or `units=(u.rad, u.rad)` or something like that).